### PR TITLE
Convert various references to SDKv2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Converts several references to SDKv1 to azure-sdk-for-go version 2.

**Which issue(s) this PR fixes**:

Refs #3409

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: azure/services/roleassignments/roleassignments.go [hunk 1]
```diff
@@ -160,9 +161,9 @@ func (s *Service) getVMSSPrincipalID(ctx context.Context) (*string, error) {
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get principal ID for VMSS")
 	}
-	resultVMSS, ok := resultVMSSIface.(compute.VirtualMachineScaleSet)
+	resultVMSS, ok := resultVMSSIface.(armcompute.VirtualMachineScaleSet)
```
Comment (nojnhuh) [inline azure/services/roleassignments/roleassignments.go:164]: Are tests failing on main without this change now that scalesets are using SDK v2?

**Review by nojnhuh — COMMENTED** [review]
/lgtm

File: azure/services/roleassignments/roleassignments.go (hunk 1 above)
Comment (mboersma) [reply azure/services/roleassignments/roleassignments.go:164]: They haven't, no, which is...interesting. Let's see if they pass here.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**